pkg/ingester/ingester.go

```diff
@@ -380,6 +380,9 @@ func New(cfg Config, clientConfig client.Config, store Store, limits Limits, con
 			logger,
 			prometheus.WrapRegistererWithPrefix("loki_", registerer))
 
+		ownedStreamsStrategy := newOwnedStreamsPartitionStrategy(i.ingestPartitionID, partitionRingWatcher, util_log.Logger)
+		i.recalculateOwnedStreams = newRecalculateOwnedStreamsSvc(i.getInstances, ownedStreamsStrategy, cfg.OwnedStreamsCheckInterval, util_log.Logger)
```

Comment:
What about this?

```go
if i.cfg.KafkaIngestion.Enabled {
    ownedStreamsStrategy := newOwnedStreamsPartitionStrategy(i.ingestPartitionID, partitionRingWatcher, util_log.Logger)
} else {
    ownedStreamsStrategy := newOwnedStreamsIngesterStrategy(i.lifecycler.ID, i.readRing, util_log.Logger)
}
i.recalculateOwnedStreams = newRecalculateOwnedStreamsSvc(i.getInstances, ownedStreamsStrategy, cfg.OwnedStreamsCheckInterval, util_log.Logger)
```

otherwise the behavior is kind of indirect. For example, `if i.recalculateOwnedStreams == nil` is true then that must also mean `if i.cfg.KafkaIngestion.Enabled` is false, which requires a bit more effort to understand.

Reply:
You're right, I think I prefer that in hindsight. I was conflicted between doing this and keeping all the kafka specific setup under one if statement. I think this is easier to understand as you said.